src/commands/Minion/alch.ts

```diff
@@ -0,0 +1,169 @@
+import hasItemEquipped from '../../lib/gear/functions/hasItemEquipped';
+import {
+	bankHasAllItemsFromBank,
+	formatDuration,
+	removeBankFromBank,
+	resolveNameBank,
+	roll
+} from '../../lib/util';
+import createReadableItemListFromBank from '../../lib/util/createReadableItemListFromTuple';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+import { AlchingActivityTaskOptions } from '../../lib/types/minions';
+import { Activity, Tasks, Time } from '../../lib/constants';
+import { rand } from '../../util';
+
+const options = {
+	max: 1,
+	time: 10_000,
+	errors: ['time']
+};
+
+const { id: bryophytasStaffId } = getOSItem("Bryophyta's staff");
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 1,
+			usage: '<quantity:int{1}|itemName:...string> [itemName:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, itemName]: [number | null | string, string]) {
+		if (typeof quantity === 'string') {
+			itemName = quantity;
+			quantity = null;
+		}
+
+		const { id: itemId, highalch, name } = getOSItem(itemName);
+		const numOfItems = await msg.author.numberOfItemInBank(itemId);
+		// 5 tick action
+		const timePerAlch = Time.Second * 3;
+
+		if (numOfItems === 0) throw `'You don't have any of this item to alch'`;
+
+		if (!quantity) {
+			quantity = numOfItems;
+		}
+
+		const maxCasts = Math.min(Math.floor(msg.author.maxTripLength / timePerAlch), numOfItems);
+
+		if (quantity * timePerAlch > msg.author.maxTripLength) {
+			throw `The max number of alchs you can do is ${maxCasts}!`;
+		}
+
+		const duration = quantity * timePerAlch;
+		let fireRuneCost = quantity * 5;
+		const natureRuneCost = quantity;
+		let savedRunes = 0;
+
+		const unlimitedFireRuneProviders = resolveItems([
+			'Staff of fire',
+			'Fire battlestaff',
+			'Mystic fire staff',
+			'Lava battlestaff',
+			'Mystic lava staff',
+			'Steam battlestaff',
+			'Mystic steam staff',
+			'Smoke battlestaff',
+			'Mystic smoke staff',
+			'Tome of fire'
+		]);
+
+		for (const runeProvider of unlimitedFireRuneProviders) {
+			if (
+				hasItemEquipped(runeProvider, msg.author.settings.get(UserSettings.Gear.Skilling))
+			) {
+				fireRuneCost = 0;
+				break;
+			}
+		}
+
+		const alchValue = quantity * highalch;
+
+		const consumedItems = resolveNameBank({
+			...(fireRuneCost > 0 ? { 'Fire rune': fireRuneCost } : {}),
+			'Nature rune': natureRuneCost,
+			[name]: quantity
+		});
+
+		const consumedItemsString = await createReadableItemListFromBank(
+			this.client,
+			consumedItems
+		);
+
+		const userBank = await msg.author.settings.get(UserSettings.Bank);
+
+		if (!bankHasAllItemsFromBank(userBank, consumedItems)) {
+			throw `You don't have the required items, you need ${consumedItemsString}`;
+		}
+
+		if (
+			hasItemEquipped(bryophytasStaffId, msg.author.settings.get(UserSettings.Gear.Skilling))
```

Comment:
since saved runes isnt used in this file, how about calculating it in the activity